website/docs/r/compute_subnetwork.html.markdown

```diff
@@ -1,98 +1,177 @@
 
-Manages a subnetwork within GCE. For more information see
-[the official documentation](https://cloud.google.com/compute/docs/vpc/#vpc_networks_and_subnets)
-and
-[API](https://cloud.google.com/compute/docs/reference/latest/subnetworks).
+A VPC network is a virtual version of the traditional physical networks
+that exist within and between physical data centers. A VPC network
+provides connectivity for your Compute Engine virtual machine (VM)
+instances, Container Engine containers, App Engine Flex services, and
+other network-related resources.
 
-## Example Usage
-
-```hcl
-resource "google_compute_subnetwork" "default-us-east1" {
-  name          = "default-us-east1"
-  ip_cidr_range = "10.0.0.0/16"
-  network       = "${google_compute_network.default.self_link}"
-  region        = "us-east1"
-}
-
-resource "google_compute_network" "default" {
-  name = "test"
-}
-```
-
-## Argument Reference
+Each GCP project contains one or more VPC networks. Each VPC network is a
+global entity spanning all GCP regions. This global VPC network allows VM
+instances and other resources to communicate with each other via internal,
+private IP addresses.
 
-The following arguments are supported:
+Each VPC network is subdivided into subnets, and each subnet is contained
+within a single region. You can have more than one subnet in a region for
+a given VPC network. Each subnet has a contiguous private RFC1918 IP
+space. You create instances, containers, and the like in these subnets.
+When you create an instance, you must create it in a subnet, and the
+instance draws its internal IP address from that subnet.
 
-* `ip_cidr_range` - (Required) The IP address range that machines in this
-    network are assigned to, represented as a CIDR block.
+Virtual machine (VM) instances in a VPC network can communicate with
+instances in all other subnets of the same VPC network, regardless of
+region, using their RFC1918 private IP addresses. You can isolate portions
+of the network, even entire subnets, using firewall rules.
 
-* `name` - (Required) A unique name for the resource, required by GCE.
-    Changing this forces a new resource to be created.
+To get more information about Subnetwork, see:
 
-* `network` - (Required) The network name or resource link to the parent
-    network of this subnetwork. The parent network must have been created
-    in custom subnet mode.
+* [API documentation](https://cloud.google.com/compute/docs/reference/rest/beta/subnetworks)
+* How-to Guides
+    * [Private Google Access](https://cloud.google.com/vpc/docs/configure-private-google-access)
+    * [Cloud Networking](https://cloud.google.com/vpc/docs/using-vpc)
 
-- - -
-
-* `description` - (Optional) Description of this subnetwork.
+## Example Usage
 
-* `project` - (Optional) The ID of the project in which the resource belongs. If it
-    is not provided, the provider project is used.
+```hcl
+  resource "google_compute_network" "custom-test" {
+    name = "%s"
```

Comment:
Let's have actual fake names instead of %s. Also, mind aligning `=`s?

Reply:
Yeah, that looks a lot better.